model/code2seq.py

```diff
@@ -1,40 +1,32 @@
-from math import ceil
-from typing import Tuple, Dict, List
+from typing import Dict, List
 
 import torch
 import torch.nn.functional as F
-from pytorch_lightning.core.lightning import LightningModule
-from torch.optim import Adam, Optimizer, SGD
-from torch.optim.lr_scheduler import ExponentialLR, _LRScheduler
-from torch.utils.data import DataLoader
 
 from configs import Code2SeqConfig
-from dataset import Vocabulary, create_dataloader, PathContextBatch
+from dataset import Vocabulary, PathContextBatch
 from model.modules import PathEncoder, PathDecoder
-from utils.common import PAD, SOS, EOS, UNK
+from utils.common import PAD, SOS
 from utils.metrics import SubtokenStatistic
+from .base_code_model import BaseCodeModel
 
 
-class Code2Seq(LightningModule):
-    def __init__(self, config: Code2SeqConfig, vocab: Vocabulary, num_workers: int = 0):
-        super().__init__()
+class Code2Seq(BaseCodeModel):
+    def __init__(
+        self, config: Code2SeqConfig, vocab: Vocabulary, num_workers: int,
+    ):
+        super().__init__(config.hyperparams, vocab, num_workers)
         self.save_hyperparameters()
-        self.config = config
-        self.vocab = vocab
-        self.num_workers = num_workers
-
-        encoder_config = self.config.encoder
-        decoder_config = self.config.decoder
         self.encoder = PathEncoder(
-            encoder_config,
-            decoder_config.decoder_size,
-            len(vocab.token_to_id),
-            vocab.token_to_id[PAD],
-            len(vocab.type_to_id),
-            vocab.type_to_id[PAD],
+            config.encoder_config,
+            config.decoder_config.decoder_size,
+            len(self.vocab.token_to_id),
+            self.vocab.token_to_id[PAD],
+            len(self.vocab.type_to_id),
+            self.vocab.type_to_id[PAD],
         )
         self.decoder = PathDecoder(
-            decoder_config, len(vocab.label_to_id), vocab.label_to_id[SOS], vocab.label_to_id[PAD]
+            config.decoder_config, len(self.vocab.label_to_id), self.vocab.label_to_id[SOS], self.vocab.label_to_id[PAD]
         )
```

Comment:
Maybe it will be better to use variables `encoder_config, decoder_config`, as it was before?